python_transport/wirepas_gateway/utils/argument_tools.py

```diff
@@ -122,75 +122,255 @@ def add_file_settings(self):
+            ),
         )
 
-        self.transport.add_argument(
-            "-u", "--username", default=None, type=str, help="MQTT broker username"
+        self.mqtt.add_argument(
+            "--mqtt_certfile",
+            default=None,
+            action="store",
+            type=str,
+            help=("Strings pointing to the PEM encoded client certificate"),
         )
 
-        self.transport.add_argument(
-            "-pw", "--password", default=None, type=str, help="MQTT broker password"
+        self.mqtt.add_argument(
+            "--mqtt_keyfile",
+            default=None,
+            action="store",
+            type=str,
+            help=(
+                "Strings pointing to the PEM "
+                "encoded client private keys "
+                "respectively"
+            ),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_cert_reqs",
+            default=ssl.CERT_REQUIRED,
+            action="store",
+            type=str,
+            help=(
+                "Defines the certificate "
+                "requirements that the client "
+                "imposes on the broker"
+            ),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_tls_version",
+            default=ssl.PROTOCOL_TLSv1_2,
+            action="store",
+            type=str,
+            help=(
+                "Specifies the version of the "
+                " SSL / TLS protocol to be used"
+            ),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_ciphers",
+            default=None,
+            action="store",
+            type=str,
+            help=(
+                "A string specifying which "
+                "encryption ciphers are allowable "
+                "for this connection"
+            ),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_persist_session",
+            default=True,
+            action="store_true",
+            help=(
+                "When False the broker will buffer "
+                "session packets between "
+                "reconnection"
+            ),
         )
 
-        self.transport.add_argument(
+        self.mqtt.add_argument(
+            "--mqtt_force_unsecure",
+            default=False,
+            action="store_true",
+            help=(
+                "When True the broker will skip "
+                "the TLS handshake"),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_allow_untrusted",
+            default=False,
+            action="store_true",
+            help=(
+                "When true the client will skip "
+                "the TLS check"),
+        )
+
+    def add_deprecated_args(self):
+        """ Deprecated mqtt arguments in order to keep backward compatibility """
+        self.deprecated.add_argument(
+            "-s",
+            "--host",
+            default=None,
+            type=str,
+            help=(
+                "Deprecated, pleas use -mqtt_hostname")
```

Comment:
I would suggest we state that this will be dropped from version 2.x onwards

Reply:
Yes, I'll update the comment